prompb/write/v2/types.proto

```diff
@@ -111,23 +113,43 @@ message Histogram {
   // Negative Buckets.
   repeated BucketSpan negative_spans =  8 [(gogoproto.nullable) = false];
   // Use either "negative_deltas" or "negative_counts", the former for
-  // regular histograms with integer counts, the latter for float
-  // histograms.
+  // regular exponential histograms with integer counts, the latter for exponential
+  // float histograms.
   repeated sint64 negative_deltas    =  9; // Count delta of each bucket compared to previous one (or to zero for 1st bucket).
   repeated double negative_counts    = 10; // Absolute count of each bucket.
 
   // Positive Buckets.
   repeated BucketSpan positive_spans = 11 [(gogoproto.nullable) = false];
   // Use either "positive_deltas" or "positive_counts", the former for
-  // regular histograms with integer counts, the latter for float
-  // histograms.
+  // regular exponential histograms with integer counts, the latter for exponential
+  // float histograms.
   repeated sint64 positive_deltas    = 12; // Count delta of each bucket compared to previous one (or to zero for 1st bucket).
   repeated double positive_counts    = 13; // Absolute count of each bucket.
 
   ResetHint reset_hint               = 14;
   // timestamp is in ms format, see model/timestamp/timestamp.go for
   // conversion from time.Time to Prometheus timestamp.
   int64 timestamp = 15;
+
+  // custom_bounds specify monotonically increasing upper inclusive boundary for
+  // the bucket counts in positive_counts or/and negative_counts (or corresponding
+  // deltas) with arbitrary widths for this histogram. For each upper boundary
+  // the previous boundary represent lower exclusive/ boundary for that bucket.
+  //
+  // The first element is the upper inclusive boundary for the first bucket,
+  // which implicitly has a lower inclusive bound of -Inf. This is similar to
+  // "le" label semantics on classic histograms. You may add a bucket with an
```

Comment:
Not sure we want to get into the question of adding a 0 boundary, that's more a question of instrumentation/visualization, feels out of scope for storage.

Reply:
Sure, just I want users to not add 0 first bound because they think they have to